en/docs/develop/injecting-parameters.md

```diff
@@ -1,84 +1,142 @@
+Micro Integrator (MI) manages configuration properties from multiple sources by following a specific order of precedence. This hierarchy is essential for understanding which configuration will take precedence when properties are duplicated across various sources.
 
-Define your synapse artifacts using "$SYSTEM:parameter_key" as the parameter value. Note that parameter_key represents a place holder representing the parameter. For example, shown below is an endpoint artifact, where the endpoint URI configured for this feature:
+- **Environment Variables:** Granted the highest precedence, these settings are specified externally, typically at the container or OS level, allowing configurations to be adjusted without code changes.
 
-```xml
-<?xml version="1.0" encoding="UTF-8"?>
-<endpoint xmlns="http://ws.apache.org/ns/synapse" name="StockQuoteEndPoint">
-  <address uri="$SYSTEM:stockQuoteEP"/>
-</endpoint>
-```
+- **System Properties:** Set at the JVM level, these are the second-highest in precedence and offer a flexible way to configure the application dynamically without altering the codebase.
+
+- **File Properties:** Residing in the `file.properties` file within the `MI_HOME/conf` directory, these have the lowest precedence and usually contain default settings that apply unless higher precedence sources override them.
+
+For example, If the `connection_name` property is defined in both an environment variable and as a system property, the value set in the environment variable will take precedence.
+
+## Initialize Configuration
 
-**Exporting the environment variable**
+The `config.properties` file, located at `PROJECT_PATH/src/main/wso2mi/resources/conf/`, plays a critical role in initializing configurable parameters. This file supports flexibility in configuration management, allowing updates to be made manually, through the `Expression Editor`, or via `Project Summary Page`. Parameters can be set as either `string` or `cert` types, with `cert` specifically used for adding a certificate file to the trust store at deployment time.
 
-In a VM deployment, you can export the environment variables as shown below. Here VAR is the URL you need to have set as environment property.
+The parameters should be defined in the config.properties file as follows:
 
-```bash
-export stockQuoteEP=http://localhost:61616/...
+```properties
+connection_name: string
+http_connection_cert: cert
 ```
 
-## Using a File
+- **Using Expression Editor**
 
-If you want to inject parameter values using a configuration file, you need to apply the following configurations.
+    1. Click on the `Ex` button on the right side of the input field to launch the Expression Editor.
+    2. Click directly inside the input field where the value needs to be entered. Alternatively, you can click on the `edit icon` positioned to the right of the `Ex` button. This will open the Expression Editor, where you can view, modify, or add new configuration values.
```

Comment:
```suggestion
    2. Click directly inside the input field where the value needs to be entered. Alternatively, you can click on the **edit** icon positioned to the right of the **Ex** button. This will open the Expression Editor, where you can view, modify, or add new configuration values.
```